octoprint_mrbeam/static/js/render_fills.js

```diff
@@ -247,89 +359,121 @@ Snap.plugin(function (Snap, Element, Paper, global) {
+                            cy,
+                            cw,
+                            ch,
+                            0,
+                            0,
+                            renderCanvas.width,
+                            renderCanvas.height
+                        );
+                    } catch (exception) {
+                        console.error(
+                            "renderCanvasContext.drawImage failed:",
+                            exception
+                        );
+                    }
+
+                    // place fill bitmap into svg
+                    const fillBitmap = renderCanvas.toDataURL("image/png");
+                    const size = getDataUriSize(fillBitmap);
+                    //                    console.debug("renderPNG rendered dataurl has " + size);
+
+                    renderCanvas.remove();
+                    return {
+                        dataUrl: fillBitmap,
+                        size: size,
+                        bbox: bbox,
+                        clusterIndex: clusterIdx,
+                    };
+                },
+                // after onerror
+                function (e) {
+                    // var len = svgDataUri ? svgDataUri.length : -1;
+                    var len = getDataUriSize(svgDataUri, "B");
+                    var msg =
+                        "Error during conversion: Loading SVG dataUri into image element failed. (dataUri.length: " +
+                        len +
+                        ")";
+                    console.error(msg, e);
+                    console.debug(
+                        "renderPNG ERR: original svgStr that failed to load: ",
+                        svgStr
+                    );
+                    console.debug(
+                        "renderPNG ERR: svgDataUri that failed to load: ",
+                        svgDataUri
+                    );
+                    new PNotify({
+                        title: gettext("Conversion failed"),
+                        text: msg,
+                        type: "error",
+                        hide: false,
+                    });
+                    if (!MRBEAM_DEBUG_RENDERING) {
+                        renderCanvas.remove();
+                    }
+                }
+            )
+            .catch(function (error) {
+                console.error(error);
             });
-        };
 
-        source.src = svgDataUri;
+        return prom;
+    };
+
+    Element.prototype.fixIds = function (selector, srcIdAttr) {
+        const root = this;
+        let elemsToFix = root.selectAll(selector);
+        for (let i = 0; i < elemsToFix.length; i++) {
+            const e = elemsToFix[i];
+            const originalId = e.attr(srcIdAttr);
+            if (originalId !== null && originalId !== "") {
+                e.attr({ id: originalId });
+            }
+            //console.log(`fixed Id: ${e.type}#${originalId}`);
+        }
```

Comment:
And what is the benefit if we rewrite and retest it this way? 
For us? For the user? At what cost?

Reply:
It's all about readability. The code is read way more often than it is being written/ rewritten, therefore readability makes it easier on ourselves.

N.B. also somewhat performance because maps can be processed faster, but that's not the issue here.